src/proxy/IReplicatedMap.ts

```diff
@@ -0,0 +1,156 @@
+import * as Promise from 'bluebird';
+import {DistributedObject} from '../DistributedObject';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+
+export interface IReplicatedMap<K, V> extends DistributedObject {
+    /**
+     * Associates a given value to the specified key and replicates it to the
+     * cluster. If there is an old value, it will be replaced by the specified
+     * one and returned from the call.
+     *
+     * @param key      key with which the specified value is to be associated.
+     * @param value    value to be associated with the specified key.
+     * @param ttl      ttl to be associated with the specified key-value pair.
+     */
+    put(key: K, value: V, ttl: number): Promise<V>;
```

Comment:
Yes, using a default value of 0 in implementation is the way to go. My comment is only for interface specification. If you specify the interface as `put(key: K, value: V, ttl?: number)`, users immediately understand that they do not have to provide a ttl value by looking at the interface.